RobotEV3/src/main/java/de/fhg/iais/roberta/visitor/codegen/Ev3C4ev3Visitor.java

```diff
@@ -1373,4 +1376,13 @@ private void generateSpeedAndPitchArguments(Expr<Void> speed, Expr<Void> pitch)
     private void generateDefaultSpeedAndPitchArguments() {
         this.sb.append("30, 50");
     }
+
+    private void generateRandomSeed() {
+        if ( this.getBean(CodeGeneratorSetupBean.class).getUsedMethods().contains(FunctionNames.RANDOM)
+            || this.getBean(CodeGeneratorSetupBean.class).getUsedMethods().contains(FunctionNames.RANDOM_DOUBLE) ) {
+            this.sb.append("srand (time(NULL));");
```

Comment:
seems that "srand (time(NULL));" is not sufficient, could you test it with an EV3?

Reply:
I don't have an EV3 but tried to run the code on simulation and it seems correct to me. Is there any problem that you have noticed with the current changes?